api-lib/circleAdmin.ts

```diff
@@ -0,0 +1,57 @@
+import assert from 'assert';
+
+import type {
+  VercelRequest,
+  VercelResponse,
+  VercelApiHandler,
+} from '@vercel/node';
+import { z } from 'zod';
+
+import { getUserFromProfileId } from './findUser';
+import { verifyHasuraAdminMiddleware } from './validate';
+
+const middleware =
+  (handler: VercelApiHandler) =>
+  async (req: VercelRequest, res: VercelResponse) => {
+    const sessionVariables = req.body.session_variables;
+    if (hasUserId(sessionVariables)) {
+      const circleId = z.number().parse(req.body.input.circle_id);
+      const profileId = z
+        .string()
+        .refine(
+          s => Number.parseInt(s).toString() === s,
+          'profileId not an integer'
+        )
+        .transform(Number.parseInt)
+        .parse(sessionVariables['x-hasura-user-id']);
+      const { role } = await getUserFromProfileId(profileId, circleId);
+      if (isNotCircleAdmin(role)) {
+        res.status(401).json({
+          message: 'User not circle admin',
+          code: 401,
+        });
+        return;
+      }
```

Comment:
```suggestion
      try {
        const { role } = await getUserFromProfileId(profileId, circleId);
        if (isNotCircleAdmin(role)) {
          res.status(401).json({
            message: 'User not circle admin',
            code: 401,
          });
          return;
        }
      }
      catch(e) {
        res.status(401).json({
          message: 'User not circle admin',
          code: 401,
        });
        return;
      }
```

Reply:
Think you might need to do a try catch here, if user doesn't exist in circle might throw an error